vertical-pod-autoscaler/pkg/apis/poc.autoscaling.k8s.io/v1alpha1/types.go

```diff
@@ -258,19 +261,25 @@ type VerticalPodAutoscalerCheckpointStatus struct {
 
 	// Checkpoint of histogram for consumption of memory.
 	MemoryHistogram HistogramCheckpoint `json:"memoryHistogram,omitempty" protobuf:"bytes,4,rep,name=memoryHistogram"`
+
+	// Timestamp of the fist sample from the histograms.
+	FirstSampleStart metav1.Time `json:"firstSampleStart,omitempty" protobuf:"bytes,5,opt,name=firstSampleStart"`
+
+	// Timestamp of the last sample from the histograms.
+	LastSampleStart metav1.Time `json:"lastSampleStart,omitempty" protobuf:"bytes,6,opt,name=lastSampleStart"`
+
+	// Total number of samples in the histograms.
+	TotalSamplesCount int `json:"totalSamplesCount,omitempty" protobuf:"bytes,7,opt,name=totalSamplesCount"`
```

Comment:
Why do we need this with decaying histograms?

Reply:
This is needed to compute confidence of the prediction. See https://github.com/kubernetes/autoscaler/pull/685 by @kgrygiel